### PR TITLE
Handle invalid MAC exception on RLPxConnectionHandler

### PR DESCRIPTION
**Pull request by ntallar**
## Description
In case incoming connections are enabled (having `server-address.interface="0.0.0.0"`), when a client is restarted with a new node key, many incoming connection attempts were received that caused both an invalid MAC and increasing the number of connected nodes well beyond the limit (ex: `Discovered 0 nodes, connected to 94/45. Trying to connect to 0 more nodes.`).

## Fix
This PR includes a fix to the huge number of connected nodes, which was caused by the [exception restarting the RLPxConnectionHandler](http://doc.akka.io/docs/akka/current/scala/fault-tolerance.html#default-supervisor-strategy), as no handling of this exception was done. This was fixed by closing the connection in case any exception happened during the MAC check.

**Note:** In the depicted scenario the invalid MAC was caused by peers trying to connect to our node using the previous node key in the RLPx handshake, as the MAC incorporates the node key, so it is an expected exception.

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandler.scala [hunk 1]
```diff
@@ -74,39 +76,51 @@ class RLPxConnectionHandler(
...
-              val size = ByteUtils.bigEndianToShort(data.take(2).toArray)
-              val (packetData, remainingData) = data.splitAt(size + 2)
-              val result = handshaker.handleResponseMessageV4(packetData)
-              processHandshakeResult(result, remainingData)
+          val maybePreEIP8Result = Try {
+            val result = handshaker.handleResponseMessage(data.take(ResponsePacketLength))
+            val remainingData = data.drop(ResponsePacketLength)
+            (result, remainingData)
+          }
+          val maybePostEIP8Result = Try {
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandler.scala]: this looks similar to line 84, is it possible to replace both with a method that returns Try?

File: src/main/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandler.scala [hunk 2]
```diff
@@ -42,9 +42,11 @@ class RLPxConnectionHandler(
 
   override def receive: Receive = waitingForCommand
 
+  def tcpActor: ActorRef = IO(Tcp)
+
   def waitingForCommand: Receive = {
     case ConnectTo(uri) =>
-      IO(Tcp) ! Connect(new InetSocketAddress(uri.getHost, uri.getPort))
+      tcpActor ! Connect(new InetSocketAddress(uri.getHost, uri.getPort))
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandler.scala]: 👍

**Comment by LukasGasior1** [comment]
Hmm that sounds like we may need supervisor strategy implemented in PeerActor, do we?
```
  override val supervisorStrategy: OneForOneStrategy =
    OneForOneStrategy() {
      case _ => Stop
    }
```
maybe something like this instead of default actor restart on exception

File: src/main/scala/io/iohk/ethereum/network/PeerActor.scala [hunk 3]
```diff
@@ -43,6 +44,11 @@ class PeerActor[R <: HandshakeResult](
   import PeerActor._
   import context.{dispatcher, system}
 
+  override val supervisorStrategy: OneForOneStrategy =
+    OneForOneStrategy() {
+      case _ => Stop
```
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: hmm actually.. we should probably also stop PeerActor here. Currently if this happens we end up with a Peer that doesn't have a child connection running. We can just use Escalate here (and PeerManager will stop the whole Peer - see its supervisorStrategy)

Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: sorry for wrong suggestion about using `Stop` :)

Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: You are right! Fixed :)

**Comment by LukasGasior1** [comment]
@ntallar needs merge

**Pull request merged**